packages/react/__tests__/get-label-from-stack-trace.js

```diff
@@ -0,0 +1,457 @@
+// @flow
+import { getLabelFromStackTrace } from '../src/emotion-element'
+
+// Ensure that it works for components that have numbers and $ in their name
+const expectedLabel = 'MyComponent-9'
+
+// All tests after this block are for runtime=automatic, i.e. the new JSX
+// transform introduced in React 17
+describe('typical function component - runtime=classic', () => {
+  test('Chrome', () => {
+    // Each "at" line starts with some whitespace
+    const stackTrace = `Error
+    at createEmotionProps (emotion-element-1fb5ab00.browser.esm.js:143)
+    at jsx (emotion-react.browser.esm.js:100)
+    at MyComponent$9 (App.js:22)
+    at renderWithHooks (react-dom.development.js:14803)
+    at mountIndeterminateComponent (react-dom.development.js:17482)
+    at beginWork (react-dom.development.js:18596)`
+
+    expect(getLabelFromStackTrace(stackTrace)).toBe(expectedLabel)
+  })
+
+  test('Firefox', () => {
+    const stackTrace = `createEmotionProps@http://localhost:3000/static/js/bundle.js:46440:40
+jsx@http://localhost:3000/static/js/bundle.js:46636:113
+MyComponent$9@http://localhost:3000/static/js/bundle.js:47600:72
+renderWithHooks@http://localhost:3000/static/js/bundle.js:18904:27
+mountIndeterminateComponent@http://localhost:3000/static/js/bundle.js:21583:13
+beginWork@http://localhost:3000/static/js/bundle.js:22697:16`
+
+    expect(getLabelFromStackTrace(stackTrace)).toBe(expectedLabel)
+  })
+
+  test('Safari', () => {
+    // Strangely, the component name does not appear in the stacktrace
```

Comment:
oh my - this is indeed weird, what's the difference between the classic and automatic runtimes here? o.O

Reply:
Yeah I have no idea LOL! I will retest this one tomorrow.